spec/models/category_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe Category, type: :model do
+  describe "Associations" do
+    it {is_expected.to belong_to(:user)}
+    it {is_expected.to have_many(:transactions).dependent(:destroy)}
+  end
+  describe "Validations" do
+    subject{FactoryBot.build(:category)}
+    context "when field name" do
+      it {is_expected.to validate_presence_of(:name)}
+      it {is_expected.to validate_length_of(:name).is_at_most(Settings.digits.length_name_max_25)}
+    end
+  end
+  describe "define enum for category_type" do
+    it {is_expected.to define_enum_for :category_type}
+  end
+  describe "Public class methods" do
+    let!(:user) {FactoryBot.create :user}
```

Comment:
tại sao chô này dùng let! e nhỉ
khác nhau giữa let và let! là gì

Reply:
`let`: khi sử dụng `let`, `variable` đó sẽ được `lazy load`, có nghĩa là chỉ khi nào nó thực sự được gọi, nó mới được tạo ra, và nó sẽ được `cache` lại cho đến khi test kết thúc.
`let!`: định nghĩa một `variable` ngay sau khi block đó được định nghĩa, được thực thi ngầm. Do đó kết quả được thực thi và `cache` lại trước khi khối it được thực thi.